knora/dsplib/models/helpers.py

```diff
@@ -339,66 +344,67 @@ def get_prefixed_iri(self, iri: Optional[str]) -> Optional[str]:
         :param iri: Fully qualified IRI
         :return: Return short from of IRI ("prefix:name")
         """
-
         if iri is None:
             return None
-        #
+
         # check if the iri already has the form "prefix:name"
-        #
         m = re.match("([\\w-]+):([\\w-]+)", iri)
         if m and m.span()[1] == len(iri):
             return iri
-        # if not self.__is_iri(iri):
+
         if not IriTest.test(iri):
-            raise BaseError("String does not conform to IRI patter: " + iri)
+            raise BaseError(f"The IRI '{iri}' does not conform to the IRI pattern.")
 
-        splitpoint = iri.find('#')
-        if splitpoint == -1:
-            splitpoint = iri.rfind('/')
-            ontopart = iri[:splitpoint + 1]
-            element = iri[splitpoint + 1:]
+        split_point = iri.find('#')
+        if split_point == -1:
+            split_point = iri.rfind('/')
+            onto_part = iri[:split_point + 1]
+            element = iri[split_point + 1:]
         else:
-            ontopart = iri[:splitpoint]
-            element = iri[splitpoint + 1:]
-        prefix = self._rcontext.get(ontopart)
+            onto_part = iri[:split_point]
+            element = iri[split_point + 1:]
+
+        prefix = self._rcontext.get(onto_part)
         if prefix is None:
-            entrylist = list(filter(lambda x: x[1].iri == ontopart, self.common_ontologies.items()))
-            if len(entrylist) == 1:
-                entry = entrylist[0]
+            entry_list = list(filter(lambda x: x[1].iri == onto_part, self.common_ontologies.items()))
```

Comment:
this could be simplified with a list comprehension

Reply:
same as above